# Выпускная квалификационная работа по курсу 'Data Science'
## Прогнозирование конечных свойств новых материалов (композиционных материалов). 

### Автор: Степанова Василиса Валерьевна

In [10]:
pip install openpyxl

     ---------------------------------------- 0.0/250.0 kB ? eta -:--:--
     -------------- ------------------------ 92.2/250.0 kB 2.6 MB/s eta 0:00:01
     -------------------------------------- 250.0/250.0 kB 2.6 MB/s eta 0:00:00
Note: you may need to restart the kernel to use updated packages.


In [4]:
# Импортирую необходимые библиотеки
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import sklearn

from sklearn.preprocessing import OrdinalEncoder
from sklearn.preprocessing import StandardScaler
from sklearn.compose import ColumnTransformer
from sklearn.model_selection import train_test_split
from sklearn.model_selection import KFold
from sklearn.model_selection import cross_val_score, cross_validate
from sklearn.model_selection import GridSearchCV
from sklearn.base import BaseEstimator
from sklearn.dummy import DummyRegressor
from sklearn.linear_model import LinearRegression
from sklearn.linear_model import Ridge
from sklearn.linear_model import Lasso
from sklearn.svm import SVR
from sklearn.neighbors import KNeighborsRegressor
from sklearn.tree import DecisionTreeRegressor
from sklearn.ensemble import RandomForestRegressor
from sklearn.ensemble import GradientBoostingRegressor
from sklearn import metrics
import pickle
import tensorflow as tf
from tensorflow import keras

In [15]:
# Загружаю исходные данные из файла X_bp
X_bp = pd.read_excel('S:\Учёба\Бауманка\ВКР\VKR_Stepanova_Vasilisa\datasets\X_bp.xlsx', index_col=0)
X_bp.shape

(1023, 10)

In [14]:
# Загружаю исходные данные из файла X_nup
X_nup = pd.read_excel('S:\Учёба\Бауманка\ВКР\VKR_Stepanova_Vasilisa\datasets\X_nup.xlsx', index_col=0)
X_nup.shape

(1040, 3)

In [16]:
# Объединяю датасеты
df = X_bp.join(X_nup, how='inner')
df.shape

(1023, 13)

В датасете осталось 1023 строки, часть строк из таблицы X_nup была отброшена.